tcc_python_scripts/tcc/wrapper.py

```diff
@@ -303,3 +307,134 @@ def active_clusters(self):
+        raw_out_folder = os.path.join(self.working_directory, 'raw_output')
+        if not os.path.isdir(raw_out_folder):
+            raise FileNotFoundError(
+                "No raw_output data, set [Output][Raw] to True and run again"
+            )
+
+        # collect the cluster names if not provided
+        if isinstance(cluster_names, type(None)):
+            cluster_name_pattern = re.compile(r'sample\.xyz.*raw_(.+)')
+            filenames = glob(
+                "{folder}/sample.xyz*raw_*".format(folder=raw_out_folder)
+            )
+            filenames = [os.path.basename(fn) for fn in filenames]
+            filenames.sort()
+            cluster_names = [
+                cluster_name_pattern.match(fn).group(1) for fn in filenames
+            ]
+        elif isinstance(cluster_names, str):
+            cluster_names = [cluster_names]
+        elif hasattr(cluster_names, '__len__'):
+            if len(cluster_names) == 0:
+                raise RuntimeError(
+                    "Empty list is illegal for cluster_names"
+                )
+        else:
+            raise TypeError("\n".join((
+                "Invalid cluster_names type, the accepted types are",
+                "\t1. None",
+                "\t2. The string of cluster name",
+                "\t3. Common container of cluster names",
+                "\t   (list, tuple, numpy array, ...)"
+            )))
+
+        # collect the result into a dict, {cluster : [frames_1, frame_2, ...]}
+        result = {}
+        for cn in cluster_names:
+            fn = glob(
+                "{folder}/sample.xyz*raw_{cluster_name}".format(
+                    folder=raw_out_folder, cluster_name=cn
+                )
+            )
+            if len(fn) == 0:
+                raise FileNotFoundError(
+                    "Raw output file for cluster {cluster} not found".format(
+                        cluster = cn
+                    )
+                )
+            else:
+                fn = fn[0]
+            is_in_cluster = xyz.get_frames_from_xyz(
+                filename=fn, usecols=[0],
+                # particle labelled as C and D were considered in the cluster
+                convert_func=lambda x : x in ['C', 'D']
+            )
+            result.update({cn : is_in_cluster})
+
+        return result
+
+    def get_cluster_table(self, cluster_names=None):
+        """
+        Getting the result of particles and the clusters they are in. The
+            example output would be like.
+
+        ..code-block::
+
+            id, FCC, 13A, 12E, 11F, 10B
+            1,    0,   0,   0,   0,   1   # particle 1 is in 10B
+            2,    1,   0,   0,   0,   0   # particle 2 is in FCC
+            3,    0,   0,   0,   0,   0   # particle 3 is not in any cluster
+            ...
+
+        For one particle, this function count all the types of clusters that this
+            particle belongs to. The result corresponds to the gross population.
```

Comment:
What does this latter paragraph in the documentation mean?

Reply:
I was trying to claim that this is not the "net" counting, but this description is a bit redundant and confusing.